waspc/data/Generator/templates/server/src/core/auth.js

```diff
@@ -69,5 +70,19 @@ export const verifyPassword = async (hashedPassword, password) => {
   }
 }
 
-export default auth
+// Looks up a user by email and if:
+// (1) the user does not exist, create them with a random password, or
+// (2) if the user does exist, update their existing password to a random password.
```

Comment:
Good thinking and question. Thankfully, once we switch to `UsernameAndPassword` we will no longer be sharing `email` and have a different method to link User accounts. 🥳 So this should not be an issue in a few days.

But yes, your scenario was correct. The reason we decided to do password reset then was because we did not have email validation yet. So a malicious user could have signed up with your email, and when you sign in with social you can get in, but so can that bad user who could use the password they used when signing up. So two people now have access, one of which you would never know about! This was determined to be a security vulnerability. But we won't have this problem thankfully. :D

Reply:
Glad to hear it! Thanks for the explanation.